unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -818,32 +815,28 @@ prettyDefinitionsBySuffixes ::
   Backend IO DefinitionDisplayResults
 prettyDefinitionsBySuffixes path root renderWidth suffixifyBindings rt codebase query = do
   hqLength <- lift $ Codebase.hashLength codebase
-  (_parseNames, printNames) <- scopedNamesForBranchHash codebase root path
-  let nameSearch :: NameSearch
-      nameSearch = makeNameSearch hqLength (NamesWithHistory.fromCurrentNames printNames)
-  DefinitionResults terms types misses <-
-    lift (definitionsBySuffixes codebase nameSearch DontIncludeCycles query)
   -- We might like to make sure that the user search terms get used as
   -- the names in the pretty-printer, but the current implementation
   -- doesn't.
-  let ppe =
-        PPE.fromNamesDecl hqLength (NamesWithHistory.fromCurrentNames printNames)
-
-      width =
+  (_parseNames, _printNames, localNamesOnly, ppe) <- scopedNamesForBranchHash codebase root path
```

Comment:
The biggest change in this function is that we previously used "prettyNames" for our initial term search which would erroneously fall-back to global names even though we're explicitly searching within a namespace.

Now any searches for scope-local things use `localNamesOnly` explicitly, and in cases where fallback is desired we use a smart PPE which does suffixification and fall-back correctly.